core/DataTable/Filter/ColumnCallbackAddMetadata.php

```diff
@@ -45,10 +45,6 @@ public function __construct($table, $columnsToRead, $metadataToAdd, $functionToA
     {
         parent::__construct($table);
 
-        if (!is_array($columnsToRead)) {
-            $columnsToRead = array($columnsToRead);
-        }
-
         $this->columnsToRead      = $columnsToRead;
```

Comment:
Found a profile from a run see screenshot. The interesting part is the exclusive metric for that function which took 2.6seconds although only called 3 times. This is a long time and got done quite a bit by this tweak but don't remember how much. Ignore the 1.3% wall time. This was before optimizing anything. Once I tweaked other parts this become like 10% or 20% wall time

![columnaddcallback](https://cloud.githubusercontent.com/assets/273120/7263255/169693e4-e8d4-11e4-87e0-2f7ab0cc7cf1.png)